### ISSUE TITLE
needAppearances() crash qrcode

### ISSUE THREAD

**Issue by NeftaliYagua** [mikehaertl/php-pdftk#182]
This is my code, see the code and results.
With needAppearances();
![qrissue](https://user-images.githubusercontent.com/2866454/76570792-1b051500-6484-11ea-8251-70e164c8a802.PNG)
![qrissue2](https://user-images.githubusercontent.com/2866454/76570803-1e000580-6484-11ea-9dba-ec890b670018.PNG)
Without needAppearances();
![after](https://user-images.githubusercontent.com/2866454/76570807-21938c80-6484-11ea-884a-63a76efa1a3a.PNG)
![after2](https://user-images.githubusercontent.com/2866454/76570808-222c2300-6484-11ea-82ce-4e751d8085e7.PNG)

**Comment by mikehaertl** [comment]
Not sure if I understand. What is the expected result?

**Comment by mikehaertl** [comment]
Now I see. I missed the issue title. I don't know how QR codes are handled in PDF. Is it already part of the PDF? How is it generated?

I'm not really an expert on PDFs but [quoting myself](https://github.com/mikehaertl/php-pdftk/issues/4#issuecomment-59006225):

>Actually "appearance" in PDF parlance means, that the PDF viewer will do whatever is neccessary to display any obscure characters that are contained in a field value. Whereas if you flatten the PDF, it means, that all the font data for non standard characters must be included in the PDF file itself (which pdftk doesn't do).

If this breaks QR codes I have no solution. I would say that you should check where the QR code is inserted and make sure that it works with appearance. We can't fix this here.

**Comment by mikehaertl** [comment]
Closing this for now as there was no further feedback and I see the problem rather with QR codes and not this library. If you think there's something to fix please provide more detail what we can to to improve the situation.

**Comment by NeftaliYagua** [comment]
I don't think there is anything to add, but something to add, possibly.

You will see I create the hidden field qr_code type extended text, and inside I place the content of the QR
Then I create a QR type object and tell it to load the content of the hidden field, if I do it this way the content works perfectly, however I cannot modify the content of the qr_code field since although I can modify it from the API the QR object it only takes the fixed content which makes it always the same although I modify it along with the other fields.
Without applying the appearance, the content still changes but the qr does not recognize it as qr but shows me the content extracted from the other hidden field, however when applying the appearance function it is already broken and fixed.

I hope I have explained myself well, it happens that it is something complex to explain.

I am sorry I did not respond in time, we have been somewhat busy with the pandemic.
